data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -107,8 +176,13 @@ public static List<DataFile> listPartition(
               .filter(FileStatus::isFile)
               .collect(Collectors.toList());
       DataFile[] datafiles = new DataFile[fileStatus.size()];
-      Tasks.Builder<Integer> task =
-          Tasks.range(fileStatus.size()).stopOnFailure().throwFailureWhenFinished();
+      Tasks.Builder<Integer> task = Tasks.range(fileStatus.size());
+
+      if (skipOnError) {
+        task.suppressFailureWhenFinished();
```

Comment:
I think it would be better to skip based on specific failures. As this is written, this would skip files when there is a transient S3 error, like a 503 "slow down". It would be better to handle errors like `ParquetDecodingException` to skip instead.

I don't think that would be an easy thing to implement, so I would say that this should at least log and should include retries for certain exceptions types.